### ISSUE TITLE
QThread instances are removed before the thread is stopped

### ISSUE THREAD

**Issue by gonzalo-bulnes** [freedomofpress/securedrop-client#1510]
## Description

When creating a thread via `QThread`, it is essential to stop the thread before destroying the `QThread` instance. Failing to do so results in crashing Python.

This behavior is visible in our functional test suite, and I've encountered it when running unit tests related to the `FileWidget` component as well (the **export** service that allows to export files runs in a separate thread).

Works towards #1513 

## Steps to Reproduce

Surprizingly enough, the functional tests on Debian 10 Buster don't report on the issue, but they do on Debian 11 Bullseye.

```sh
make venv
source .venv/bin/activate
# Running a functional test is enough. (I specified a test and a random order seed, but that's probably unnecessary.)
python -m pytest -v --random-order-bucket global tests/functional/test_delete_source.py --random-order-seed=305816
```

## Expected Behavior

Python exits with status code `0`.

## Actual Behavior

Python prints an error message and exits with status code `134`. (Which I _believe_ corresponds to `SIGABRT` - process aborted due to critical error in the underlying C code, I might be wrong on that - https://stackoverflow.com/a/62022759.)

```sh
make venv
source .venv/bin/activate
python -m pytest -v --random-order-bucket global tests/functional/test_delete_source.py --random-order-seed=305816

# [...] regular test output
# 
# QThread: Destroyed while thread is still running
# Aborted

echo $?
# 134
```

## Comments

I've already looked into this issue as part of the "Export All Files", and I know how to manage the threads (example: [show the diff of this file](https://github.com/freedomofpress/securedrop-client/commit/249cc873c2b286e2ce548b85f14fc4a9cdaa6fb2#diff-b9f8828b97a9e5a9772e99886228db6659dd993060e33ecbeda19ffbdad310f9R13-R39)).

As far as I can tell, there are four `QThread` instances currently in use: one "main" thread, one for "sync" purposes, one for "download" purposes and one for "export" purposes. Those could be created when the application is started and managed there (because that's where the exit signals are handled), then injected as dependencies to the components that use them.

**Comment by gonzalo-bulnes** [comment]
I labelled this issue as "Bullseye". While this is not Bullseye-specific, it does become more visible, and may be part of the current difficulties we have running the functional test suite.

**Comment by gonzalo-bulnes** [comment]
@creviera Oooh, there were two aspects to this issue, and I didn't capture well the _functional_ tests suite aspect here.

Arguably, the functional tests aspect is the one I know for sure was `QThread`-related (they were both segmentation faults, that's where it gets confusing) and I should have written this issue description more in line with that.

I'm hesitating to re-open it, and have it closed by #1512 - but I can do that on Monday if that helps. The PR is already linked to this issue, it's just that in hindsight I shouldn't have set #1510 to close it immediately. :slightly_smiling_face:
